### PR TITLE
[SPARK-34784][BUILD] Upgrade Jackson to 2.12.2

### PR DESCRIPTION
**Pull request by xingchaozh**
### What changes were proposed in this pull request?

This pr upgrade Jackson to 2.12.2. 
Jackson Release 2.12: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.12

### Why are the changes needed?

Make it easy to upgrade Avro 1.10.2.

```
[error] Caused by: sbt.ForkMain$ForkError: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.11.4 requires Jackson Databind version >= 2.11.0 and < 2.12.0
[error] 	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
[error] 	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
[error] 	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Tested with Avro 1.10.2 and Parquet 1.12.0: https://github.com/apache/spark/runs/2157735537

### CODE REVIEW

**Comment by wangyum** [comment]
ok to test.

File: dev/deps/spark-deps-hadoop-2.7-hive-2.3 [hunk 1]
```diff
@@ -104,17 +104,16 @@ httpclient/4.5.13//httpclient-4.5.13.jar
...
-jackson-databind/2.11.4//jackson-databind-2.11.4.jar
-jackson-dataformat-yaml/2.11.4//jackson-dataformat-yaml-2.11.4.jar
+jackson-core/2.12.2//jackson-core-2.12.2.jar
+jackson-databind/2.12.2//jackson-databind-2.12.2.jar
+jackson-dataformat-yaml/2.12.2//jackson-dataformat-yaml-2.12.2.jar
 jackson-datatype-jsr310/2.11.2//jackson-datatype-jsr310-2.11.2.jar
 jackson-jaxrs/1.9.13//jackson-jaxrs-1.9.13.jar
 jackson-mapper-asl/1.9.13//jackson-mapper-asl-1.9.13.jar
-jackson-module-jaxb-annotations/2.11.4//jackson-module-jaxb-annotations-2.11.4.jar
-jackson-module-paranamer/2.11.4//jackson-module-paranamer-2.11.4.jar
```
Comment (wangyum) [inline dev/deps/spark-deps-hadoop-2.7-hive-2.3:116]: `jackson-module-paranamer` removed by https://github.com/FasterXML/jackson-module-scala/commit/b71f8dc54ff8b926dea3db92012136e8fcd83bde

**Review by wangyum — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
cc @srowen FYI

**Comment by srowen** [comment]
Jenkins test this please

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, All. Please hold on this. Apache Avro and Apache Parquet and Jackson update are related.
- https://github.com/apache/spark/pull/31866#issuecomment-802935725

**Comment by wangyum** [comment]
Jenkins, retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. @wangyum shows the integration result: This PR(Jackson 2.12.2), Avro(1.10.2), Parquet(1.12.0).
- https://github.com/apache/spark/pull/31649

cc @iemejia

**Comment by dongjoon-hyun** [comment]
The failure is irrelevant.
```
[info] - SPARK-34757: should ignore cache for SNAPSHOT dependencies *** FAILED *** (412 milliseconds)
[info]   0 equaled 0 (SparkSubmitUtilsSuite.scala:321)
```

**Comment by wangyum** [comment]
Merged to master.

**Pull request closed without merging**